lime/api/api.py

```diff
@@ -147,7 +150,35 @@ def post_vertex(vertex_type):
     doc = Vertex()
     doc.vertex_type = vertex_type
     doc.host = Host.by_current_user()
-    data = {k: request.json[k] for k in doc.get_aggregate_fields() if k in request.json.keys()}
+
+    data = {}
+
+    # Get the field which is an OpenGraphURLField, or None
+    # There should only be one! (Or, more often, none)
+    # If this is throwing an exception, you might have more than one in the
+    # CustomVertex schema
+    og_url_field, = doc.get_typed_fields(OpenGraphURLField) or [None]
+
+    # If there is an OpenGraphURLField, get it and add the data to the vertex
+    if og_url_field:
+        og_url = request.json[og_url_field]
+        og_fields = { # Prefix keys with og:
+            'og:{}'.format(k): v
+            for k, v in opengraph.OpenGraph(url=og_url).items()
```

Comment:
It doesn't say anything about waiting or synchronous execution in the docs. 

So, if its not asynchronous, does execution block until the request comes back?